src/main.c

```diff
@@ -6046,6 +6089,9 @@ static long ctrl_ioctl(struct file *filp, unsigned int cmd, unsigned long arg){
 			break;
 		}
 
+		if (snap_devices[minor])
+			elastio_snap_wait_for_release(snap_devices[minor]);
```

Comment:
Complete all read operations: this resolves an additional bug with -EBUSY during the transition to incremental.